packages/eslint-plugin/src/rules/no-unsafe-member-access.ts

```diff
@@ -71,7 +71,8 @@ export default util.createRule({
     }
 
     return {
-      MemberExpression: checkMemberExpression,
+      // ignore MemberExpression if it's parent is TSClassImplements or TSInterfaceHeritage
+      'MemberExpression:not(TSClassImplements > MemberExpression):not(TSInterfaceHeritage > MemberExpression)': checkMemberExpression,
```

Comment:
`:not(...)` accepts any valid selector as its argument, which means I believe you can simplify this slightly by merging everything into one:
```suggestion
      ':not(TSClassImplements, TSInterfaceHeritage) > MemberExpression': checkMemberExpression,
```

Reply:
Nice, that is simpler and tests still pass. I've added the change.